pkg/webhook/hook.go

```diff
@@ -0,0 +1,218 @@
+package webhook
+
+import (
+	"context"
+	"encoding/json"
+	"io/ioutil"
+	"net/http"
+
+	"github.com/pkg/errors"
+	admission "k8s.io/api/admission/v1"
+	admissionregistration "k8s.io/api/admissionregistration/v1"
+	corev1 "k8s.io/api/core/v1"
+	k8serrors "k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/types"
+	"k8s.io/client-go/util/cert"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/manager"
+	logf "sigs.k8s.io/controller-runtime/pkg/runtime/log"
+
+	v1 "github.com/percona/percona-xtradb-cluster-operator/pkg/apis/pxc/v1"
+	"github.com/percona/percona-xtradb-cluster-operator/pkg/pxctls"
+)
+
+const certPath = "/tmp/k8s-webhook-server/serving-certs/"
+
+var hookPath = "/validate-pxc-percona-com"
+
+type hook struct {
+	cl        client.Client
+	caBunlde  []byte
+	namespace string
+}
+
+func (h *hook) Start(i <-chan struct{}) error {
+	err := h.createWebhook()
+	if err != nil {
+		logf.Log.Info("Can't create webhook", "error", err.Error())
+	}
+	<-i
+	return nil
+}
+
+func (h *hook) createWebhook() error {
+	failPolicy := admissionregistration.Fail
+	sideEffects := admissionregistration.SideEffectClassNone
+	hook := &admissionregistration.ValidatingWebhookConfiguration{
+		ObjectMeta: metav1.ObjectMeta{
+			Name: "pxc-validation-webhook",
+		},
+		Webhooks: []admissionregistration.ValidatingWebhook{
+			{
+				AdmissionReviewVersions: []string{"v1", "v1beta1"},
+				Name:                    "validationwebhook.pxc.percona.com",
+				ClientConfig: admissionregistration.WebhookClientConfig{
+					Service: &admissionregistration.ServiceReference{
+						Namespace: h.namespace,
+						Name:      "percona-xtradb-cluster-operator",
+						Path:      &hookPath,
+					},
+					CABundle: h.caBunlde,
+				},
+				SideEffects:   &sideEffects,
+				FailurePolicy: &failPolicy,
+				Rules: []admissionregistration.RuleWithOperations{
+					{
+						Rule: admissionregistration.Rule{
+							APIGroups:   []string{"pxc.percona.com"},
+							APIVersions: []string{"*"},
+							Resources:   []string{"*/*"},
+						},
+						Operations: []admissionregistration.OperationType{"CREATE", "UPDATE"},
+					},
+				},
+			},
+		},
+	}
+	err := h.cl.Create(context.TODO(), hook)
+	if err != nil {
+		if k8serrors.IsAlreadyExists(err) {
+			err = h.cl.Delete(context.TODO(), hook)
```

Comment:
is it possible to use the `Update` command instead?
```suggestion
			return h.cl.Update(context.TODO(), hook)
```